lib/extensions/ar_pglogical_migration.rb

```diff
@@ -0,0 +1,27 @@
+module ArPglogicalMigration
+  def migrate(direction)
+    PglogicalSubscription.all.each do |s|
+      provider_region = MiqRegion.find(s.provider_region)
+
+      while !provider_region.migrations_ran.include?(version) do
+        s.disable
```

Comment:
I'm trying understand the purpose of  3 lines. Why do we need to disable, enable the subscription and then sleep here?

Reply:
This is to ensure that the apply process "wakes up" at every migration to consume "queued" changes from the remote region. Really we only need to do this if the subscription is "down" so I can make this conditional.